src/index.ml

```diff
@@ -106,6 +106,7 @@ module Make (K : Key) (V : Value) (IO : IO) = struct
     may Bloomf.clear t.entries;
     Tbl.clear t.log_mem;
     may (fun i -> IO.clear i.io) t.index;
+    (match t.index with None -> () | Some index -> IO.close index.io);
```

Comment:
This could be pushed into the `may` function above, which is there to handle exactly this type of pattern :slightly_smiling_face: